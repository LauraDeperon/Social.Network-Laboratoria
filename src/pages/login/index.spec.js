import { Login } from './index.js';
import * as service from '../../services/index.js';
import * as util from '../../utils/history.js';

service.SignIn = jest.fn(() => Promise.resolve(true));
service.signInGoogle = jest.fn(() => Promise.resolve(true));
util.onNavigate = jest.fn(() => Promise.resolve(true));

describe('Login', () => {
  it('should be a function', () => {
    expect(typeof Login).toBe('function');
  });
  it('should render the page', () => {
    expect(typeof Login).toMatchSnapshot();
  });
  it('should call SignIn when the user click the button', () => {
    Login().querySelector('#signin-button').dispatchEvent(new Event('click'));
    return expect(service.SignIn).toHaveBeenCalled();
  });
  it('Sign in error', async () => {
    Login().querySelector('#signin-button').dispatchEvent(new Event('click'));
    try {
      await service.SignIn;
    } catch (error) {
      expect(error).toThrow(error);
    }
  });
  it('should call signInGoogle when the user click the button', () => {
    Login().querySelector('#signingoogle-button').dispatchEvent(new Event('click'));
    return expect(service.signInGoogle).toHaveBeenCalled();
  });
  it('Sign in with Google error', async () => {
    Login().querySelector('#signingoogle-button').dispatchEvent(new Event('click'));
    try {
      await service.signInGoogle;
    } catch (error) {
      expect(error).toThrow(error);
    }
  });
  it('should redirect the user to register page', () => {
    Login().querySelector('#signup-button').dispatchEvent(new Event('click'));
    return expect(util.onNavigate).toHaveBeenCalledWith('/register');
  });
});
