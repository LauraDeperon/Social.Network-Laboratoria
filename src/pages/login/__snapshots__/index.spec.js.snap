// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should render the page 1`] = `"function"`;
